skimage/data/__init__.py

```diff
@@ -49,6 +54,159 @@
            'shepp_logan_phantom',
            'stereo_motorcycle']
 
+legacy_data_dir = osp.abspath(osp.dirname(__file__))
+skimage_distribution_dir = osp.join(legacy_data_dir, '..')
+
+try:
+    import pooch
+    from pooch.utils import file_hash
+    has_pooch = True
+except ImportError:
+    has_pooch = False
+    import hashlib
+
+    def file_hash(fname):
+        # Versioned (i.e. copied from) from pooch.utils
+        # Calculate the hash in chunks to avoid overloading the memory
+        chunksize = 65536
+        hasher = hashlib.sha256()
+        with open(fname, "rb") as fin:
+            buff = fin.read(chunksize)
+            while buff:
+                hasher.update(buff)
+                buff = fin.read(chunksize)
+        return hasher.hexdigest()
+
+
+def _fetch(data_filename):
+    # potential_local_file is the location of the file that would be shipped
+    # with scikit-image.
+    #
+    # Two things can cause the file to already be in the distribution:
+    # 1. The user is running it from the github repo, in which case
+    #    we should avoid a downloading from the internet
+    # 2. We chose to ship this file with skimage by default
+    potential_local_file = osp.join(skimage_distribution_dir, data_filename)
+    resolved_path = osp.join(data_dir, '..', data_filename)
```

Comment:
The variable naming here is a bit confusing.  Can we perhaps go for something like `gh_repository_path` and `cache_path` to make it clear what is being referred to?

The logic below I'd also perhaps write some one-liner functions to make intent clear:

```
if in_cache(filename) and has_hash(filename, expected_hash):
    ...
```

Reply:
Thanks for the guidance. Please see rework.